src/builtins/builtins_cd.c

```diff
@@ -0,0 +1,67 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   builtins_cd.c                                      :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: yliu <yliu@student.42.jp>                  +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/09/04 20:37:58 by yliu              #+#    #+#             */
+/*   Updated: 2024/09/17 19:27:01 by yliu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "builtins.h"
+
+static int	move_to_home(void)
+{
+	char	*home_dir;
+	int		ret;
+
+	home_dir = getenv("HOME");
+	ret = chdir(home_dir);
+	if (ret != 0)
+		return (ret);
+	return (0);
+}
+
+static char	*join_path(const char *cwd, const char *dirname)
+{
+	int		i;
+	char	*tmp;
+	char	*tmp2;
+
+	i = 0;
+	while (cwd[i])
+		i++;
+	if (i == 0)
+		return (ft_xstrdup(dirname));
+	else if (cwd[i - 1] == '/')
+		tmp = ft_xstrdup(cwd);
+	else
+		tmp = ft_xstrjoin(cwd, "/");
+	tmp2 = ft_xstrjoin(tmp, dirname);
+	free(tmp);
+	return (tmp2);
+}
+
+int	builtins_cd(char **args, t_builtins_ctx *config)
```

Comment:
Since type of variable `config` is `t_builtins_ctx`, should it be named `ctx`?

``` suggestion
int	builtins_cd(char **args, t_builtins_ctx *ctx)
```